service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/models/BasicDigitalTwin.java

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.sdk.iot.service.digitaltwin.models;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An optional, helper class for deserializing a digital twin.
+ * Only properties with non-null values are included.
+ */
+public final class BasicDigitalTwin {
+    @JsonProperty(value = "$dtId", required = true)
+    private String id;
+
+    @JsonProperty(value = "$metadata", required = true)
+    private DigitalTwinMetadata metadata;
+
+    @JsonIgnore
+    private final Map<String, Object> customProperties = new HashMap<>();
+
+    /**
+     * Gets the unique Id of the digital twin in a digital twins instance. This field is present on every digital twin.
+     * @return The unique Id of the digital twin in a digital twins instance. This field is present on every digital twin.
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Sets the unique Id of the digital twin in a digital twins instance. This field is present on every digital twin.
+     * @param id The unique Id of the digital twin in a digital twins instance. This field is present on every digital twin.
+     * @return The BasicDigitalTwin object itself.
+     */
+    private BasicDigitalTwin setId(String id) {
+        this.id =id;
+        return this;
```

Comment:
we should check if other models in our repo are fluent, if not, then these shouldn't have fluent setters either. We'd need to be consistent across this library.